specification/hugr.md

```diff
@@ -90,7 +90,20 @@ represent (typed) data or control dependencies.
 
 ## Functional description
 
-A HUGR is a directed graph with nodes and edges. The nodes represent
+A HUGR is a directed graph. There are several different types of node, and
+several different types of edge, with different semantics, described below.
+
+Nodes usually have additional data associated with them.
+
+Edge types are of three kinds:
```

Comment:
I think it is confusing to use the term "kind" confusing here since we use `EdgeKind` later. I suggest "Connectivity". 

I also don't think this is a good place to discuss this, it would work better under Edges? I think I would introduce the concept of connectivity and immediately state the connectivity of each kind of edge, at the bottom of the Edges section (after describing the kinds)

Reply:
Yeah I'm not sure about that simple example, nearly removed it last time, I guess it works as a "soon you will understand what all of this means!" teaser.

"Category" works, but given we don't need to talk about it again, I think it's just simpler to add a section on "Ports" under "Edges" and describe the port/ordering constraints of each edge kind there.

Reply:
Have created a subsection "Ports and ordering of edges" and rejigged the headings below.